src/index.ts

```diff
@@ -74,78 +83,93 @@ app.post("/", async (req, res, next) => {
 
       const { mevshare, flashbotsBundleProvider } = await initClients(provider, verifiedSignatureSearcherPkey);
 
-      // If configured, simulate the original bundle to check if it reverts without the unlock.
-      if (env.passThroughNonReverting) {
-        const originalSimulationResponse = await flashbotsBundleProvider.simulate(backrunTxs, targetBlock);
-        if (!originalBundleReverts(originalSimulationResponse, req)) {
-          await handleUnsupportedRequest(req, res, "Original bundle does not revert"); // Pass through if the original bundle doesn't revert.
-          return;
+      let bundle: BundleParams["body"], refunds: Refund[];
+      // If ovalHeaderConfigs.unlockAddresses are configured, send the unlock transaction bundles and the backrun bundle without simulation.
+      // This setting enables the searcher to request a specific list of unlock addresses for use in their bundle and
+      // accelerates the process by omitting the step of finding unlock addresses and performing simulations.
+      if (ovalHeaderConfigs) {
+        const ovalAddresses = ovalHeaderConfigs.unlockAddresses.map((unlockAddress) => unlockAddress.ovalAddress);
+        const { unlockBundles } = await getUnlockBundlesFromOvalAddresses(
+          flashbotsBundleProvider,
+          backrunTxs,
+          targetBlock,
+          ovalAddresses,
+          req,
+        );
+
+        Logger.debug(
+          req.transactionId,
+          "Header unlock addresses found. Sending unlock tx bundle and backrun bundle...",
+          ovalAddresses,
+        );
+        // Construct the outer bundle with the modified payload to backrun the UnlockLatestValue call.
+        bundle = [
+          ...unlockBundles,
+          ...backrunTxs.map((tx): { tx: string; canRevert: boolean } => {
+            return { tx, canRevert: false };
+          }),
+        ];
+
+        // Calculate refunds for the each unlock bundles this assumes the unlock bundles are the first elements in the bundle
+        // and the ovalAddresses are in the same order as the unlock bundles.
+        refunds = calculateBundleRefunds(ovalAddresses, ovalConfigs);
+
+        return sendBundle(req, res, mevshare, targetBlock, body.id, bundle, refunds);
+      } else {
```

Comment:
this else seems redundant since you always return from the prev if

Reply:
This is the case where valid oval headers are not present, so this code is always executed in that scenario.